domain/src/main/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputHasNoFractionalPartRuleClassifierProvider.kt

```diff
@@ -24,7 +24,6 @@ internal class FractionInputHasNoFractionalPartRuleClassifierProvider @Inject co
     )
   }
 
-  // TODO(#210): Add tests for this classifier.
   override fun matches(answer: Fraction): Boolean {
```

Comment:
@rt4914 I'm confused here. Why doesn't this class have any input?

Reply:
@MohamedMedhat1998 
```
internal class FractionInputHasNoFractionalPartRuleClassifierProvider @Inject constructor(
  private val classifierFactory: GenericRuleClassifier.Factory
) : RuleClassifierProvider, GenericRuleClassifier.NoInputInputMatcher<Fraction> {

  override fun createRuleClassifier(): RuleClassifier {
    return classifierFactory.createNoInputClassifier(
      InteractionObject.ObjectTypeCase.FRACTION,
      this
    )
  }

  override fun matches(answer: Fraction): Boolean {
    return answer.numerator == 0
  }
}
```

Reply:
@prayutsu according to the Java docs of this class  
```
Provider for a classifier that determines whether a fraction has no fractional part per the fraction input interaction.
```
Which means there should be some sort of input. That is what is confusing me.

Reply:
https://github.com/oppia/oppia/blob/37285a/extensions/interactions/FractionInput/directives/fraction-input-rules.service.ts#L58

Line number 95 in https://github.com/oppia/oppia-android/blob/develop/domain/src/main/java/org/oppia/android/domain/classify/rules/GenericRuleClassifier.kt

Reply:
@MohamedMedhat1998 I am not able to understand your exact question here? I mean what are you expecting?

To me this looks correct because the purpose of this classifier is to just check if the inputted answer contains fractions part of not. It has nothing to do with the correctness of answer.

Reply:
I was only confused with the term `answer` vs the term `input`, but I got what you mean related to this classifier. Everything is clear now in terms of this PR.